core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -244,6 +244,28 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
       leaderId = members.keys.headOption
   }
 
+  def maybeElectNewJoinedLeader(): Option[MemberMetadata] = {
+    leaderId match {
+      case Some(currentLeaderId) =>
+        val oldLeader = get(currentLeaderId)
```

Comment:
nit: this reads a bit odd. Wouldn't it make more sense to call this `currentLeader`?

Reply:
Sounds good!